src/mud/base.py

```diff
@@ -113,31 +333,312 @@ def fit(self, **kwargs):
         # e.g. to comptue E(r) and to pass on to future iterations
         self._r = np.divide(self._ob, self._pr)
 
-        # If present, multiply by weights (e.g. from a previous iteration)
-        if self._weights is not None:
-            self._r *= self._weights
-
         # Multiply by initial to get updated pdf
-        up_pdf = np.multiply(self._in, self._r)
+        up_pdf = np.multiply(self._in * self._weights, self._r)
         self._up = up_pdf
 
     def mud_point(self):
+        """Maximal Updated Density (MUD) Point
+
+        Returns the Maximal Updated Density or MUD point as the parameter
+        sample from the initial distribution with the highest update density
+        value:
+
+        .. math::
+            \\lambda^{MUD} := \\text{argmax} \\pi_{up}(\\lambda)
+            :label: mud
+
+        Note if the updated distribution has not been computed yet, this
+        function will call :meth:`fit` to compute it.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        mud_point : np.ndarray
+            Maximal Updated Density (MUD) point.
+        """
         if self._up is None:
             self.fit()
         m = np.argmax(self._up)
         return self.X[m, :]
 
     def estimate(self):
+        """Estimate
+
+        Returns the best estimate for most likely paramter values for the
+        given model data using the data-consistent framework.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        mud_point : ndarray
+            Maximal Updated Density (MUD) point.
+        """
         return self.mud_point()
 
+    def exp_r(self):
```

Comment:
thoughts on calling it `ratio_expected`? would show up next to `ratio` in the doc autocomplete.

Reply:
`expected_ratio` is better even.